cnf-certification-test/observability/suite.go

```diff
@@ -151,49 +151,68 @@ func testTerminationMessagePolicy(env *provider.TestEnvironment) {
 	testhelper.AddTestResultLog("Non-compliant", failedContainers, tnf.ClaimFilePrintf, ginkgo.Fail)
 }
 
-func checkPDBIsValid(pdb *policyv1.PodDisruptionBudget, replicas *int32) (bool, error) {
-	var replicaCount int32
-	if replicas != nil {
-		replicaCount = *replicas
-	} else {
-		replicaCount = 1 // default value
-	}
-
-	if pdb.Spec.MinAvailable != nil && pdb.Spec.MinAvailable.IntValue() == 0 {
-		return false, fmt.Errorf("field .spec.minAvailable cannot be zero")
-	}
-
-	if pdb.Spec.MaxUnavailable != nil && pdb.Spec.MaxUnavailable.IntValue() >= int(replicaCount) {
-		return false, fmt.Errorf("field .spec.maxUnavailable cannot be greater than or equal to the number of pods in the replica")
-	}
-
-	return true, nil
-}
-
+//nolint:funlen
 func testPodDisruptionBudgets(env *provider.TestEnvironment) {
-	failedPDBs := []string{}
-	for i := range env.PodDisruptionBudgets {
-		pdb := &env.PodDisruptionBudgets[i]
-		for pdbLabelKey, pdbLabelValue := range pdb.Spec.Selector.MatchLabels {
-			// Go through all deployments and statefulsets, objects for which PDBs apply
-			for _, deployment := range env.Deployments {
-				if deployment.Spec.Template.Labels[pdbLabelKey] == pdbLabelValue {
-					if ok, err := checkPDBIsValid(pdb, deployment.Spec.Replicas); !ok {
-						failedPDBs = append(failedPDBs, pdb.Name)
-						tnf.ClaimFilePrintf("PDB %s is not valid for deployment %s, err: %v", pdb.Name, deployment.Name, err)
+	var compliantObjects []*testhelper.ReportObject
+	var nonCompliantObjects []*testhelper.ReportObject
+
+	// Loop through all of the of Deployments and StatefulSets and check if the PDBs are valid
+	for _, d := range env.Deployments {
+		// Check if there are zero PDBs, if so this is a failure.
+		if len(env.PodDisruptionBudgets) == 0 {
+			nonCompliantObjects = append(nonCompliantObjects, testhelper.NewReportObject("FAIL: Deployment is missing a corresponding PodDisruptionBudget", testhelper.DeploymentType, false).
```

Comment:
This change should have a non-backward-compatible warning in the next release's notes.